src/api/scan/base_controller.go

```diff
@@ -112,11 +112,57 @@ func NewController() Controller {
 				return "", errors.Errorf("configuration option %s not defined", cfg)
 			}
 		},
+		metadata: func(r *scanner.Registration) (*v1.ScannerAdapterMetadata, error) {
```

Comment:
As the controller has been moved to here, does this callback func still need? I think it can all the scanner manager to get the scanner metadata.